### ISSUE TITLE
tmt tool supports command line options for setting test repo

### ISSUE THREAD

**Issue by psklenar** [psss/tmt#140]
tmt tool supports command line options for setting test repo
Without this test repo has to be set manually in the ci.fmf plan, or additional workaround script is required.

so
 tmt plans create ci2 --template full --repository https://github.com/psklenar/tmt
will create :
summary:
    Essential command line features
discover:
    how: fmf
    **repository: https://github.com/psklenar/tmt**
prepare:
    how: ansible
    playbooks: plans/packages.yml
execute:
    how: beakerlib

**Comment by psklenar** [comment]
I am going to try create merge request about this ...

**Comment by psss** [comment]
Thanks for the idea. Looking forward to the pull request.

**Comment by thrix** [comment]
@psklenar if you need any help/pointers to the implementation, we have each Tuesday hacking sessions, invited you!

**Comment by psklenar** [comment]
@thrix suggest to add:
**tmt plans update**
which updates some elements in NAME.fmf

**Comment by psklenar** [comment]
or https://src.osci.redhat.com/rpms/lftp/blob/private-fmf-plan/f/ci.fmf

**Comment by psss** [comment]
When thinking about this a little bit more I realized we could also support multiple values for each `discover` step option. This would only require to always provide the same set of options:

    tmt plan create plans/basic \
        --name upstream --how fmf --url https://github/ --ref devel --filter tier:0,1 \
        --name fedora --how fmf --url https://fedora/ --ref master --filter tier:0 \
        --name rhel --how fmf --url https://rhel/ --ref rhel-8 --filter tier:1

An empty string could be used for undefined options if needed like this:

    tmt plan create plans/basic \
        --name upstream --how fmf --url https://github/ --ref devel --filter tier:0,1 \
        --name fedora --how fmf --url https://fedora/ --ref '' --filter ''
        --name rhel --how fmf --url https://rhel/ --ref rhel-8 --filter tier:1

I guess this would also be a little bit easier from the implementation point of view. What do you think?

**Comment by mruprich** [comment]
This sounds good, I will start working on a PR for you on this.

**Comment by thrix** [comment]
@psss would it make simpler to have a `tmt plan update` command which could update the plan with multiple discover phases, but it would be a separate command?

So your

```
tmt plan create plans/basic \
    --name upstream --how fmf --url https://github/ --ref devel --filter tier:0,1 \
    --name fedora --how fmf --url https://fedora/ --ref '' --filter ''
    --name rhel --how fmf --url https://rhel/ --ref rhel-8 --filter tier:1
```
Would translate to

```
tmt plan create plans/basic --name upstream --how fmf --url https://github/ --ref devel --filter tier:0,1 \
tmt plan update plans/basic discover --name fedora --how fmf --url https://fedora/ --ref '' --filter ''
tmt plan update plans/basic discover --name rhel --how fmf --url https://rhel/ --ref rhel-8 --filter tier:1
```
or something like that?

**Comment by psss** [comment]
My impression is that adjusting existing plans in an automated way will not be such a common use case. I would expect that further changes of the existing fmf data will be done in most cases manually. Plus, from the implementation point of view, it's not that easy to update fmf node. It's quite easy when plan is stored in a separate file but thanks to [elasticity](https://fmf.readthedocs.io/en/latest/features.html#elasticity) you can also have multiple plans in a single file, and [inheritance](https://fmf.readthedocs.io/en/latest/features.html#inheritance) does not make this task easy either. Not that it would be impossible but for now I think we're good with extending the `tmt plan create` options.

**Comment by psss** [comment]
@mruprich, any progress here? Is there anything blocking you? Can I help somehow?

**Comment by mruprich** [comment]
No, sorry, I just got stuck on other issues. Hopefully I will have the solution ready tmrw.

**Comment by psss** [comment]
Fixed by #219. See the [documentation](https://tmt.readthedocs.io/en/latest/examples.html#create-plans) for usage examples.
